shared-bindings/typing/__init__.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * This file is part of the Micro Python project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * SPDX-FileCopyrightText: Copyright (c) 2022 Dan Halbert for Adafruit Industries
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+//| """Typing features module
+//|
+//| This module is intended to be a compatible subset of `Python's module
+//| of the same name <https://docs.python.org/3.10/library/typing.html>`_
+//| """
+//|
+//| TYPE_CHECKING: bool = False
+//| """
+//| indicates that expensive type checks should be skipped.
+//|
+//| In CircuitPython using the exact following sequence prevents the body
+//| of the 'if' statement (including the qstr identifiers it uses)
+//| from being included in the mpy file at all::
+//|
+//|     from typing import TYPE_CHECKING
+//|     if TYPE_CHECKING:
+//|         import module_may_not_even_exist_in_circuitpython
+//|
+//| Together with ``from __future__ import typing`` this can help
```

Comment:
`from __future__ import typing` does not exist in CPython. I don't think we should add such an incompatibility.

Reply:
yes, should have been `from __future__ import annotations